lib/src/layer/attribution_layer.dart

```diff
@@ -1,11 +1,104 @@
+    super.key,
+    required Widget Function(BuildContext context) customBuilder,
+  })  : _customBuilder = customBuilder,
+        attributions = null,
+        alignment = Alignment.bottomRight,
+        backgroundColor = const Color(0xCCFFFFFF),
+        animationDuration = const Duration(milliseconds: 250),
+        animationCurve = Curves.fastOutSlowIn;
+
+  const AttributionLayer({
+    super.key,
+    this.attributions,
+    this.alignment = Alignment.bottomRight,
+    this.backgroundColor = const Color(0xCCFFFFFF),
+    this.animationDuration = const Duration(milliseconds: 250),
+    this.animationCurve = Curves.fastOutSlowIn,
+  }) : _customBuilder = null;
+
+  final Widget Function(BuildContext context)? _customBuilder;
+
+  final Map<Text, void Function()>? attributions;
+  final Alignment alignment;
+  final Color backgroundColor;
+
+  final Duration animationDuration;
+  final Curve animationCurve;
+
+  @override
+  State<AttributionLayer> createState() => _AttributionLayerState();
+}
+
+class _AttributionLayerState extends State<AttributionLayer>
+    with TickerProviderStateMixin {
+  late final _animationController = AnimationController(
+    duration: widget.animationDuration,
+    vsync: this,
+  );
+  late final _animation = CurvedAnimation(
+    parent: _animationController,
+    curve: widget.animationCurve,
+  );
+
+  late final Map<Text, void Function()> _attributions = {
+    ...widget.attributions ?? {},
+    const Text('flutter_map & the authors'): () {},
+  };
+
+  @override
+  Widget build(BuildContext context) => widget._customBuilder != null
+      ? widget._customBuilder!(context)
+      : Align(
+          alignment: widget.alignment,
+          child: ColoredBox(
+            color: widget.backgroundColor,
+            child: Padding(
+              padding: const EdgeInsets.symmetric(vertical: 4, horizontal: 6),
+              child: GestureDetector(
+                onTap: () {
+                  if (_animation.status != AnimationStatus.completed) {
+                    _animationController.forward();
+                  } else {
+                    _animationController.reverse();
+                  }
+                },
+                child: IntrinsicWidth(
+                  child: Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      SizeTransition(
+                        sizeFactor: _animation,
+                        axis: Axis.vertical,
+                        child: SingleChildScrollView(
+                          child: Column(
+                            crossAxisAlignment: CrossAxisAlignment.end,
+                            children: [
+                              ..._attributions.keys.toList(),
+                              const Divider(),
+                            ],
+                          ),
+                        ),
+                      ),
+                      const Text('Show Map Attributions'),
```

Comment:
A few thoughts on this:

1. Extremely minor nitpick, but I think this can be shortened to just "Attributions" if it's going to be pure text; it should be obviously hyperlink styled, right?
2. Any objections to an info image? This is pretty standard practice and could look better if done right.
3. I don't know if this is easily doable, but the ideal solution IMO is to do something like Leaflet and collapse into in info button only if the text doesn't fit. If this is to difficult though, something like this would be great.

I may have some other suggestions about the UX after giving it a try later, but these are my thoughts off the top of my head.

Reply:
Yeah I'm still not sure on the best way to do this. At the moment, the attributions expand upwards, with each one on a new line. This doesn't look great.
If you want it to look a specific way (which I have no problem with, open to all ideas) can you just post a very rough sketch from something like Paint, just so we can review that before spending time implementing it.

Maybe a 50% opacity flutter_map logo (the same one on the website, and I still have the original design files if it needs modifying) and 50% opacity info icon will look ok. When tapped, it can open a floating box of some sort? Additionally this would add appropriate space for an image attribution like Mapbox requires. What do you think of this?